framework/aster-frame/src/trap/handler.rs

```diff
@@ -182,3 +183,62 @@ cpu_local! {
 pub fn in_interrupt_context() -> bool {
     IN_INTERRUPT_CONTEXT.load(Ordering::Acquire)
 }
+
+fn handle_kernel_page_fault(f: &TrapFrame) {
+    // We only create mapping: `vaddr = paddr + PHYS_OFFSET` in kernel page fault handler.
+    let page_fault_vaddr = x86_64::registers::control::Cr2::read().as_u64();
+    debug_assert!(is_kernel_vaddr(page_fault_vaddr as usize));
+
+    // Check kernel region
+    // FIXME: The modification to the offset mapping of the kernel code and data should not permitted.
+    debug_assert!({
+        let kernel_region = MemoryRegion::kernel();
+        let start = kernel_region.base();
+        let end = start + kernel_region.base() + kernel_region.len();
+        !((start..end).contains(&(page_fault_vaddr as usize)))
```

Comment:
I think that the page fault should _not_ be handled unless we have _verified_ that handling it is correct. So we should never use `debug_assert` here. At least we should use `assert`, and even explicitly using `panic` might be worthwhile.

This also means that, we should check the conditions under which handling page fault _is_ correct (e.g., the page fault address falls in some range), not exclude the situations where handling page faults is _not_ correct (e.g., the page fault address falls out of some range).

In the following code, this argument actually appears in the comments: `Safety: The page fault address has been checked`. However, the statement is simply not true since `debug_assert` is used here.

I would suggest adding a constant `PHYS_MEM_CAP_VADDR`, which is `PHYS_MEM_BASE_VADDR + N TiB`, and checking that the page fault address actually falls within this range.

(Note that in the Linux kernel, the kernel text/data mapping is isolated from the direct mapping of all physical memory, but I have not checked whether Asterinas has followed this design.)